core/lib/src/data/data.rs

```diff
@@ -124,6 +127,10 @@ impl Data {
         // Set the read timeout to 5 seconds.
         net_stream.set_read_timeout(Some(Duration::from_secs(5))).expect("timeout set");
 
+        // Grab the certificate info
```

Comment:
All of this `tls` stuff should be grouped together to avoid using so many `feature(tls)` attributes.

One idea is to have a block:

```rust
#[cfg(feature = "tls")]
{
    let mut data = Data::new(http_stream);
    if let Some(certs) = net_stream.get_peer_certificates() {
        data.set_peer_certificates(certs);
    }

    Ok(data)
}

#[cfg(not(feature = "tls"))]
Ok(Data::new(http_stream))
```